src/components/Navbar/SearchDrawer/SearchDrawer.tsx

```diff
@@ -53,46 +59,38 @@ const SearchDrawer: React.FC = () => {
     }
   }, [isOpen, focusInput]);
 
-  // This useEffect is triggered when the debouncedSearchQuery value changes
-  // eslint-disable-next-line react-func/max-lines-per-function
+  /**
+   * Call BE to fetch the results using the passed filters.
+   *
+   * @param {string} query
+   * @param {number} page
+   */
+  const getResults = useCallback(
+    (query: string, page: number) => {
+      searchGetResults(
+        SearchQuerySource.SearchDrawer,
+        query,
+        page,
+        PAGE_SIZE,
+        setIsSearching,
+        setHasError,
+        setSearchResult,
+        router.query.languages as string,
```

Comment:
This needs to be discussed with Osama as it makes more sense to use the current locale of the entire app than a query param. I know the search page uses this query param but for the search drawer, most probably it won't.